compiler/src/iree/compiler/GlobalOptimization/FuseHorizontalContractions.cpp

```diff
@@ -0,0 +1,424 @@
+  reassoc.push_back({0, 1});
+  for (int i = 0, e = rhsType.getRank() - 1; i < e; ++i) {
+    reassoc.push_back({i + 2});
+  }
+
+  SmallVector<int64_t> rhsNewShape(rhsType.getShape());
+  rhsNewShape.insert(rhsNewShape.begin(), 1);
+  RankedTensorType concatRhsType =
+      RankedTensorType::get(rhsNewShape, rhsType.getElementType());
+
+  SmallVector<Value> rhsVals;
+  for (auto op : fusionGroup.contractionOps) {
+    Value thisRhs = op.getDpsInputOperand(1)->get();
+    Value expanded = rewriter.create<tensor::ExpandShapeOp>(loc, concatRhsType,
+                                                            thisRhs, reassoc);
+    rhsVals.push_back(expanded);
+  }
+
+  Value newRhs = rewriter.create<tensor::ConcatOp>(loc, 0, rhsVals);
+
+  SmallVector<int64_t> newShape(outType.getShape());
+  newShape.insert(newShape.begin(), rhsVals.size());
+  RankedTensorType concatOutType =
+      RankedTensorType::get(newShape, outType.getElementType());
+
+  Value baseOut = base.getDpsInitOperand(0)->get();
+  auto origFill = baseOut.getDefiningOp<linalg::FillOp>();
+  auto origEmpty =
+      origFill.getDpsInitOperand(0)->get().getDefiningOp<tensor::EmptyOp>();
+
+  auto newEmpty = rewriter.create<tensor::EmptyOp>(loc, concatOutType,
+                                                   origEmpty.getDynamicSizes());
+  auto newFill = rewriter.create<linalg::FillOp>(
+      loc, concatOutType, origFill.getDpsInputOperand(0)->get(),
+      newEmpty.getResult());
+
+  Value lhs = base->getOperand(0);
+
+  SmallVector<AffineMap> indexingMaps = base.getIndexingMapsArray();
+  SmallVector<utils::IteratorType> iteratorTypes = base.getIteratorTypesArray();
+  iteratorTypes.insert(iteratorTypes.begin(), utils::IteratorType::parallel);
+
+  indexingMaps[0] = indexingMaps[0].shiftDims(1);
+  indexingMaps[1] = indexingMaps[1].shiftDims(1).insertResult(
+      rewriter.getAffineDimExpr(0), 0);
+  indexingMaps[2] = indexingMaps[2].shiftDims(1).insertResult(
+      rewriter.getAffineDimExpr(0), 0);
+
+  linalg::GenericOp newGenericOp = rewriter.create<linalg::GenericOp>(
+      loc, concatOutType, ValueRange{lhs, newRhs}, newFill.getResult(0),
+      indexingMaps, iteratorTypes);
+  rewriter.cloneRegionBefore(base->getRegion(0), newGenericOp.getRegion(),
+                             newGenericOp.getRegion().begin());
+
+  Value fusedResult = newGenericOp.getResult(0);
+  if (fusionGroup.truncateOps) {
+    // Insert truncate operator.
+    auto truncType =
+        baseTruncOp.value()->getResult(0).getType().cast<RankedTensorType>();
+    auto concatTruncType =
+        RankedTensorType::get(newShape, truncType.getElementType());
+    Value truncOuts = rewriter.create<tensor::EmptyOp>(
+        loc, concatTruncType, origEmpty.getDynamicSizes());
+    SmallVector<AffineMap> truncIndexingMaps(
+        2, AffineMap::getMultiDimIdentityMap(concatTruncType.getRank(),
+                                             rewriter.getContext()));
```

Comment:
nit: this can be `rewriter.getMultiDimIdentityMap(concatTruncType.getRank())`, which looks cleaner.

[optional] can we declare a variable (e.g., `rank = concatTruncType.getRank()`), and use it here and below? It makes code shorter and cleaner.